Dockerfile

```diff
@@ -1,5 +1,4 @@
-FROM quay.io/openshift/origin-must-gather:4.12.0 as builder
-
+FROM registry.ci.openshift.org/ocp/4.16:cli as builder
```

Comment:
Hi @msherif1234 , I'm from the multiarch OCP team and i'm reaching as I @memodi wanted to disable some test cases while you are working on doing this enablement. 

This image is not multiarch either.

The alternatives for you are:
- use `cli-artifacts`, copy from /usr/share and keep the one binary you actually need:

```dockerfile

FROM registry.ci.openshift.org/ocp/4.16:cli-artifacts as builder

FROM runtime

COPY --from=builder /usr/share/openshift /tmp/openshift

RUN mv "/tmp/openshift/linux_$(uname -m | sed 's/aarch64/arm64/;s/x86_64/amd64/')/oc" /usr/bin/oc && rm -rf /tmp/openshift
```

```
.
├── LICENSE
├── linux_amd64
│   ├── oc
│   ├── oc.rhel8
│   └── oc.rhel9
├── linux_arm64
│   ├── oc
│   ├── oc.rhel8
│   └── oc.rhel9
├── linux_ppc64le
│   ├── oc
│   ├── oc.rhel8
│   └── oc.rhel9
├── linux_s390x
│   └── oc
├── mac
│   └── oc
├── mac_arm64
│   └── oc
└── windows
    └── oc.exe

```


- Do not use a first stage, but download the cli from mirror.openshift.org similar to https://github.com/openshift-qe/baremetal-qe-infra/blob/master/images/prow-image/Dockerfile#L16C1-L22C1:
```Dockerfile
RUN set -x; \
    OC_TAR_URL="https://mirror.openshift.com/pub/openshift-v4/$(uname -m)/clients/ocp/latest/openshift-client-linux.tar.gz" && \
    curl -L -q -o /tmp/oc.tar.gz "$OC_TAR_URL" && \
    tar -C /usr/bin/ -xvf /tmp/oc.tar.gz oc && \
    ln -sf /usr/bin/oc /usr/bin/kubectl && \
    rm -f /tmp/oc.tar.gz
```


- Use the multiarch payload as a base. For example, `quay.io/openshift-release-dev/ocp-release:4.15.12-multi`

The cli image is:
```shell
└ $ oc adm release info --image-for=cli quay.io/openshift-release-dev/ocp-release:4.15.12-multi
quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
```

```Dockerfile
# image-for=cli quay.io/openshift-release-dev/ocp-release:4.15.12-multi
FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:... as builder

FROM runtime

COPY --from=builder /usr/bin/oc /usr/bin/oc
```

Reply:
Hi @aleskandro 
I tried the first option but its not working when I try to collect MG
I see the following errors
```sh
[must-gather-bkdlg] POD 2024-05-06T16:51:35.500707795Z /usr/bin/oc: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /usr/bin/oc)
[must-gather-bkdlg] POD 2024-05-06T16:51:35.500738588Z /usr/bin/oc: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /usr/bin/oc)
[must-gather-bkdlg] POD 2024-05-06T16:51:35.500758901Z /usr/bin/oc: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /usr/bin/oc)
```

Reply:
2nd option seems to work on x86 will check on another arch

Reply:
@msherif1234 depending on the base image of the runtime stage image, you can need to use oc.rhel8.

I said rhel8 as I see cs8. It is going to be EOL soon, why don't you bump to cs9?